src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -38,33 +77,59 @@ public static SocksBeginExBuilder beginEx()
         return new SocksBeginExBuilder();
     }
 
+
     public static final class SocksRouteExBuilder
     {
-        private final SocksRouteExFW.Builder routeExRw;
+        private final SocksRouteExFW.Builder routeExRW;
 
         private SocksRouteExBuilder()
         {
             MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
-            this.routeExRw = new SocksRouteExFW.Builder().wrap(writeBuffer, 0, writeBuffer.capacity());
+            this.routeExRW = new SocksRouteExFW.Builder().wrap(writeBuffer, 0, writeBuffer.capacity());
         }
 
         public SocksRouteExBuilder address(
-            String address)
+            String address) throws UnknownHostException
         {
-            this.routeExRw.address(address);
+            if (IPV4_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final Matcher ipv4Matcher = IPV4_ADDRESS_MATCHER.get();
+                final byte[] ipv4AddressBytes = IPV4_ADDRESS_BYTES.get();
+                encodeIpv4AddressBytes(ipv4Matcher, ipv4AddressBytes);
+                routeExRW.address(b -> b.ipv4Address(s -> s.set(ipv4AddressBytes)));
+            }
+            else if (IPV6_STD_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final byte[] addressBytes = IPV6_ADDRESS_BYTES.get();
+                final Matcher ipv6Matcher = IPV6_STD_ADDRESS_MATCHER.get();
+                encodeStandardIpv6AddressBytes(ipv6Matcher, addressBytes);
+                routeExRW.address(b -> b.ipv6Address(s -> s.set(addressBytes)));
+            }
+            else if (IPV6_HEX_COMPRESSED_VALIDATE_MATCHER.get().reset(address).matches() &&
+                IPV6_HEX_COMPRESSED_MATCHER.get().reset(address).matches())
```

Comment:
Please reverse the order, and add a test passing more than 8 groups as a candidate IPv6 address.
This raises the question of whether or not all address strings that are _not_ valid IPv4 or IPv6 are valid domain names.
Looks like we need another regular expression, this time for domain names.
See https://wiki.namecoin.org/?title=Domain_Name_Specification_2.0#Regular_Expression for details.